### PR TITLE
Run tidy if installed, fail if it's not and we're in travis

### LINKED ISSUE #189: Make the tidy command optional
tidy is used by the test suite, but we don't want to ask all developers to install it.

We do want to fail if it's not installed in CI, though (if `TRAVIS` environment variable is `true`).

See https://github.com/ocaml/odoc/pull/185#issuecomment-421617855.

cc @ostera

### PR DESCRIPTION
**Pull request by ostera**
Close #189.

### CODE REVIEW

**Comment by ostera** [comment]
I can use `Unix.open_process_full` to capture all the output?

**Comment by aantron** [comment]
Thank you!

File: test/html/test.ml [hunk 1]
```diff
@@ -198,7 +205,12 @@ let () =
     (* Running the actual commands for the test. *)
     let run_test_case () =
       generate_html ();
-      validate_html html_file;
+
+      match running_in_travis, found_tidy with
+        | true, false -> Alcotest.fail "Could not find `tidy` in PATH!";
+        | _, true -> validate_html html_file;
+        | _, _ -> ();
```
Comment (rizo) [inline test/html/test.ml]: FYI, this currently breaks all the tests locally.

In particular when the matched pattern is `false, true` (second line), only the HTML validation will be run. This has to do with the OCaml's precedence rules. The content after this line is included in the last case after `(); ...`.

Comment (rizo) [reply test/html/test.ml]: Maybe we should consider running ocamlformat on odoc :)

**Pull request merged**